flink-end-to-end-tests/flink-kafka-test-base/src/main/java/org/apache/flink/kafka/tests/KafkaEvent.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.kafka.tests;
+
+/**
+ * The event type used in the {@link Kafka010Example}.
+ *
+ * <p>This is a Java POJO, which Flink recognizes and will allow "by-name" field referencing
+ * when keying a {@link org.apache.flink.streaming.api.datastream.DataStream} of such a type.
+ * For a demonstration of this, see the code in {@link Kafka010Example}.
+ */
+public class KafkaEvent {
```

Comment:
I have talked with @tzulitai and we concluded that to avoid this duplication it would be best to:

- Move submodule (and rename) `flink-kafka-test-base` to `flink-streaming-examples/flink-kafka-example-base`
- Kafka version specifics (`flink-streaming-examples/flink-kafka-example` and `flink-streaming-examples/flink-kafka-0.10-example`) extend that base
- End to end tests just use the example jobs for testing

Reply:
OK, it sounds a good solution.

Reply:
@pnowojski I think currently there is not a module named `flink-streaming-examples` but a module named `flink-examples-streaming` which is not a parent module (and contains many kind of examples). Do you mean that I need to create a module called `flink-streaming-examples`?

Reply:
No, sorry about a typo. I was thinking about creating a submodule in the existing one `flink-examples/flink-examples-streaming/`

Reply:
@pnowojski and @aljoscha It is hoped that the end-to-end testing for the modern Kafka connector can be merged before the 1.7.0 release.

Reply:
I think the new flat organisation that you proposed is fine, makes it more fine grained and easier to understand/re-use by the user.